102-concurrency/goroutines/race-condition-pointers-fix-with-atomic/race-condition-pointers-fix-with-atomic.go

```diff
@@ -0,0 +1,31 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+	"sync/atomic"
+)
+
+type RaceTest struct {
+	Val int32
+}
+
+func main() {
+	raceTest := &RaceTest{}
+
+	wg := &sync.WaitGroup{}
+	wg.Add(10000)
+
+	for i:=0; i<10000; i++ {
+		go increment(raceTest, wg)
+	}
+
+	wg.Wait()
+
+	fmt.Println(raceTest)
+}
+
+func increment(rt *RaceTest, wg *sync.WaitGroup) {
+	atomic.AddInt32(&rt.Val, 1)
+	wg.Done()
+}
```

Comment:
atomic ornegi guzel olmus, aynisini farkli baska bir fonksiyonda mutex ile ornegini yapip aralarinda benchmark uyguladigimiz bir test eklesek nasil olurdu?

Reply:
Çok güzel bir örnek olur aslında, ikisi arasındaki performans farkını görmüş oluruz hem, 

https://github.com/GoTurkiye/training/pull/7 şuradaki pr da mutex ile çözümü var. Her ikisi merge edildiğinde yeni bir branch üzerinden benchmark testlerini yazarsak temiz olur diye düşünüyorum?

Reply:
👍 istersen o PR'daki commiti cherry-pick ile bu PR'a tasiyip islemleri buradan da devam ettirebiliriz tek bir .go dosyasi uzerinden. ne dusunursun?

Reply:
Selamlar, ilgili pr merge edildi, buradaki süreci ilerletebilirsiniz.

Reply:
> istersen o PR'daki commiti cherry-pick ile bu PR'a tasiyip islemleri buradan da devam ettirebiliriz tek bir .go dosyasi uzerinden. ne dusunursun?

102-concurrency/goroutines altında bir dizin oluşturalım o zaman, #7 deki merge ile gelen mutex lock çözümünü içerisine ekleyip benchmark testleri için tek bir dosya açarız. İlk söylediğin gibi tek bir dosyada 2 farklı metodu da eklediğimizde güzel olacaktır. Ne dersin? 

En sonda dizinin görüntüsü şu şekilde olacak. 

![Screenshot 2021-09-29 at 17 22 04](https://user-images.githubusercontent.com/26525468/135287841-cb9592c2-e493-432c-9a64-54a1c94150e4.png)